org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/backend/SonarLintEclipseHeadlessRpcClient.java

```diff
@@ -190,6 +190,13 @@ public void didSynchronizeConfigurationScopes(Set<String> configurationScopeIds)
     allAffectedConnections.forEach(c -> c.getAndCacheAllSonarProjects());
   }
 
+  @Override
+  public boolean matchProjectBranch(String configurationScopeId, String branchNameToMatch, SonarLintCancelChecker cancelChecker)
+    throws ConfigScopeNotFoundException {
+    var project = SonarLintUtils.resolveProject(configurationScopeId);
+    return branchNameToMatch.equals(VcsService.matchSonarProjectBranch(project, branchNameToMatch, Set.of(branchNameToMatch)));
```

Comment:
FYI: This is a re-used call from the `VcsService`: We provide the branch and it wants to also have all the available branches (in our case there is only one) to find the best one.
If there is none, `null` is returned which is of course not equal to `branchNameToMatch`